synapse/exc.py

```diff
@@ -31,6 +31,9 @@ def get(self, name, defv=None):
         '''
         return self.errinfo.get(name, defv)
 
+    def set(self, name, valu):
```

Comment:
This should be named setdefault() if it's not going to hulksmash values

Reply:
( but i like the idea of having a `set()` API too! )

Reply:
changed this to just hulksmash the value in.